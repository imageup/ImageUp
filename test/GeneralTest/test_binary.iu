function main() -> int
{
	print(1 + 2);
	print(3 - 2);
	print(4 * 2);
	print(5 / 2);
	printf(1.0 + 2.3);
	printf(4.0 * 3.2);
	printb(4 > 3);
	printb(3 < 4);
	printb(7.0 <= 23.4);
	printb(7.0 >= 23.4);
	printb( 4 == 2);
	
	bool: b1 = true;
	bool: b2 = true;
	if (b1 && b2){
		prints("and works");
	}

	bool: b3 = false;
	if (b1 || b3){
		prints("or works");
	}


	return 0;
}