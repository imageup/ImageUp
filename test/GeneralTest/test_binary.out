3
1
8
2
3.3
12.8
1
1
1
0
0
and works
or works
